sbt-version-policy/src/main/scala/sbtversionpolicy/DependencyCheckReport.scala

```diff
@@ -1,42 +1,44 @@
 package sbtversionpolicy
 
 import coursier.version.{ModuleMatchers, Version, VersionCompatibility}
-import dataclass.data
-import lmcoursier.definitions.{ModuleMatchers => _, _}
+import lmcoursier.definitions.{ModuleMatchers => *, *}
 
-@data class DependencyCheckReport(
-  backwardStatuses: Map[(String, String), DependencyCheckReport.ModuleStatus],
-  forwardStatuses: Map[(String, String), DependencyCheckReport.ModuleStatus]
+case class DependencyCheckReport(
+  binaryCompatibilityReport: Map[(String, String), DependencyCheckReport.ModuleStatus],
+  sourceCompatibilityReport: Map[(String, String), DependencyCheckReport.ModuleStatus]
 ) {
-  def validated(direction: Direction): Boolean =
-    (!direction.backward || backwardStatuses.forall(_._2.validated)) &&
-      (!direction.forward || forwardStatuses.forall(_._2.validated))
 
-  def errors(direction: Direction, ignored: Set[(String, String)] = Set.empty): (Seq[String], Seq[String]) = {
+  def validated(incompatibilityType: IncompatibilityType): Boolean =
+    incompatibilityType match {
+      case IncompatibilityType.BinaryIncompatibility => binaryCompatibilityReport.forall(_._2.validated)
+      case IncompatibilityType.SourceIncompatibility => sourceCompatibilityReport.forall(_._2.validated)
+    }
+
+  def errors(incompatibilityType: IncompatibilityType, ignored: Set[(String, String)] = Set.empty): (Seq[String], Seq[String]) = {
 
-    val backwardElems =
-      if (direction.backward) backwardStatuses else Map()
-    val forwardElems =
-      if (direction.forward) forwardStatuses else Map()
+    val relevantErrors =
+      incompatibilityType match {
+        case IncompatibilityType.BinaryIncompatibility => binaryCompatibilityReport
+        case IncompatibilityType.SourceIncompatibility => sourceCompatibilityReport
+      }
```

Comment:
I thought the switch statements added a bit of unnecessary line count, I've put a tweak into https://github.com/julienrf/sbt-version-policy/pull/1 that you're welcome to accept if you like!